app/src/main/java/com/example/recipeopedia/activities/FavoriteRecipeDetailsActivity.java

```diff
@@ -24,19 +32,26 @@
 import com.example.recipeopedia.models.Review;
 import com.parse.FindCallback;
 import com.parse.ParseException;
+import com.parse.ParseFile;
 import com.parse.ParseQuery;
 import com.parse.ParseUser;
 import com.parse.SaveCallback;
 
 import org.parceler.Parcels;
 
+import java.io.File;
 import java.util.ArrayList;
 import java.util.List;
 
 public class FavoriteRecipeDetailsActivity extends AppCompatActivity {
     public static final String TAG = "FavoriteRecipeDetailsActivity";
+    public static final String KEY_ATTEMPT_PICTURE = "attemptPicture";
+    public final static int CAPTURE_IMAGE_ACTIVITY_REQUEST_CODE = 42;
     private FavoriteRecipe favoriteRecipe;
-    private Button btnViewAttempts;
+    private Button btnViewAttempts, btnCaptureImage, btnUploadImage;
+    private ImageView ivRecipeImage;
+    public String photoFileName = "photo.jpg";
```

Comment:
@JayArri mentioned this as well - you need a unique file name to prevent collisions, you could, for example, concatenate "photo" with `System.currentTimeMillis()`